cpp/cmake/thirdparty/get_arrow.cmake

```diff
@@ -90,8 +161,12 @@ function(find_and_configure_arrow VERSION BUILD_STATIC ENABLE_S3 ENABLE_ORC ENAB
   endif()
 
   rapids_cpm_find(
-    Arrow ${VERSION}
-    GLOBAL_TARGETS arrow_shared parquet_shared arrow_dataset_shared
+    Arrow
+    ${VERSION}
+    # TODO: Should we set the list of global targets conditionally based on whether we're building
+    # shared/static and whether parquet is enabled?
```

Comment:
This mostly feels like a cleanup task, I don't think it's super important

Reply:
You can have a superset of targets for all configurations in the global target set as only those that exist will be promoted. It was design for this use-case of not wanting to handle configuration permutations/combinations